### DISCUSSION TITLE
Allow users to add and edit data (apps, and shortcuts)

### DISCUSSION THREAD

**Discussion by gabimor** [gabimor/keyboardninja#22] [Ideas]
In order for KeyboardNinja.me to be really useful, we need to allow users to add missing apps and missing shortcuts easily.
We want to allow even less tech savvy users the opportunity to contribute, so how would such an app/shortcut editor look ?

Requirements:
- allow users to enter shortcuts easily or maybe import them from other resources.
- allow users to update existing shortcuts. its important to keep the existing shortcuts ids since they are linked to users who starred them. (so we can't just recreate the entire app with new keys for every update)

Questions:
- what sort of user changes are allowed ? edits to shortcuts ? deletes ?
- do we need to approve user changes ?
- how to support bulk edit and granular edits ? have a fancy UI or a simple text (YAML) file import ?

**Comment by ShragaUser** [comment]
Just a thought, I would love a shortcut editor similar to what vs-code gives you when configuring shortcuts. 
Sort of a "type and I'll show you a SVG of the keys as you press them". And also I'm pretty sure we can leverage existing open source to do this in this app. 

Regarding approving - I think we need some sort of validation. I was thinking of starting off simple with an admin page and simple yes/no on "suggestions". 
Maybe later incorporate something the like of a "PR" where a review is required (and maybe the owner of the existing shortcut can approve changes?).

**Reply by gabimor** [reply]
Having a UI oriented editor is cool for adding a few shortcuts, but what if I wanna add all the shortcuts for photoshop I scraped from adobe ? I was thinking about a light txt style file in a simple format even non technicals can understand (kind of YAML?) you can upload or paste in to KBN, all we just need to validate.
regarding approvals, I agree, start as simple as possible @ShragaUser
